components/knowledge/FileModal.tsx

```diff
@@ -0,0 +1,84 @@
+import {
+  Modal,
+  ModalBody,
+  ModalContent,
+  useDisclosure,
+} from '@nextui-org/react';
+import { Image } from '@nextui-org/image';
+import { BiPlus } from 'react-icons/bi';
+import { NotionFileModal } from '@/components/knowledge/Notion';
+import { isNotionConfigured, runNotionSync } from '@/actions/knowledge/notion';
+import { useState } from 'react';
+
+interface FileModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  handleAddFile: () => void;
+}
+
+const FileModal = ({ isOpen, onClose, handleAddFile }: FileModalProps) => {
+  const notionModal = useDisclosure();
+  const [isSyncing, setIsSyncing] = useState(false);
+  const [notionConfigured, setNotionConfigured] = useState(false);
+
+  const onClickNotion = async () => {
+    onClose();
+    notionModal.onOpen();
+    const isConfigured = await isNotionConfigured();
+    if (!isConfigured) {
+      setIsSyncing(true);
+      await runNotionSync(false);
+      setIsSyncing(false);
```

Comment:
this is going to load indefinitely if `runNotionSync` throws an error

Reply:
Good call, will fix 👍